engine/src/main/java/org/terasology/engine/particles/components/ParticleEmitterComponent.java

```diff
@@ -110,4 +110,26 @@
      * Allows checking only some particles each update since it's a heavy operation.
      */
     public int collisionUpdateIteration;
+
+    @Override
+    public void copy(ParticleEmitterComponent other) {
+        this.maxParticles = other.maxParticles;
+        this.particleCollision = other.particleCollision;
+        this.spawnRateMax = other.spawnRateMax;
+        this.spawnRateMin = other.spawnRateMin;
+        this.enabled = other.enabled;
+        this.lifeTime = other.lifeTime;
+        this.particleSpawnsLeft = other.particleSpawnsLeft;
+        this.destroyEntityWhenDead = other.destroyEntityWhenDead;
+        this.ownerEntity = other.ownerEntity;
+        this.particlePool = other.particlePool;
+        this.generatorFunctionMap.clear();
+        this.generatorFunctionMap.putAll(other.generatorFunctionMap);
+        this.affectorFunctionMap.clear();
+        this.affectorFunctionMap.putAll(other.affectorFunctionMap);
+        this.locationComponent = new LocationComponent();
+        this.locationComponent.copy(other.locationComponent); // TODO check this
```

Comment:
This likely works, but I think to remember that @immortius advised us to not have direct references between components. So the "real fix" here would be to change the implementation of the emitter component accordingly.